src/daemons/rhsmcertd.c

```diff
@@ -627,6 +703,11 @@ parse_cli_args (int *argc, char *argv[])
             exit (EXIT_FAILURE);
         }
     }
+
+    // When --debug CLI option is set, then override configuration options
+    if (show_debug) {
+        log_level = LOG_LEVEL_DEBUG;
+    }
```

Comment:
i think this is redundant: in `main()`, right after the call to `parse_cli_args()` (this function) `get_config()` is called, which will read the configuration bits; since we want `-d`/`--debug` to override what's in the configuration, then the same code later on in `main()` after `get_config()` will do the same job already

Reply:
Maybe we would like to see debug messages during reading configuration file.

Reply:
This needs a better comment then, otherwise it is not clear why it is done in this place. This also means it is possible to avoid the second "copy" of this code in `main()` by simply limiting the reading of the config bits for debug in `key_file_init_config()` within a `if (!show_debug) { ... }` block: they are only useful only in case `-d`/`--debug` is not specified anyway

Reply:
Good point. Done